src/rust/aggregator/src/pipelines/update_materialized_view.rs

```diff
@@ -0,0 +1,66 @@
+use anyhow::anyhow;
+use chrono::{DateTime, Duration, Utc};
+use sqlx::PgPool;
+
+use aggregator::{Pipeline, PipelineAggregationResult, PipelineError};
+
+pub struct UpdateMaterializedView {
+    pool: PgPool,
+    view_name: String,
+    update_interval: std::time::Duration,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+#[allow(dead_code)]
+impl UpdateMaterializedView {
+    pub fn new(
+        pool: PgPool,
+        view_name: impl Into<String>,
+        update_interval: std::time::Duration,
+    ) -> Self {
+        Self {
+            pool,
+            view_name: view_name.into(),
+            update_interval,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Pipeline for UpdateMaterializedView {
+    fn model_name(&self) -> String {
+        String::from("UpdateMaterializedView")
+    }
+
+    fn ready(&self) -> bool {
+        self.last_indexed_timestamp.is_none()
+            || self.last_indexed_timestamp.unwrap()
+                + Duration::from_std(self.update_interval).unwrap()
+                < Utc::now()
+    }
+
+    async fn process_and_save_historical_data(&mut self) -> PipelineAggregationResult {
+        self.process_and_save_internal().await
+    }
+
+    fn poll_interval(&self) -> Option<std::time::Duration> {
+        Some(self.update_interval)
+    }
+
+    async fn process_and_save_internal(&mut self) -> PipelineAggregationResult {
+        sqlx::query(
```

Comment:
Do you need to run SQLx prepare for this?

Reply:
No. This is unchecked. Using `query!` doesn't work here, since it doesn't look like it supports passing table/views names as an argument. And it also doesn't support passing a processed string as an argument, it wants a string literral.